@xen-orchestra/web-core/lib/components/ui/input/UiInput.vue

```diff
@@ -1,141 +1,179 @@
-<!-- WIP -->
+<!-- v5 -->
 <template>
-  <div class="ui-input">
-    <VtsIcon :icon accent="current" class="before" />
-    <input :id v-model.trim="modelValue" class="typo p1-regular input" type="search" v-bind="attrs" />
-    <VtsIcon
-      v-if="!attrs.disabled && modelValue"
-      :icon="faXmark"
-      class="after"
-      accent="info"
-      @click="modelValue = ''"
-    />
+  <div class="ui-input" :class="toVariants({ accent })">
+    <UiLabel v-if="slots.default || label" :accent="labelAccent" :required :icon="labelIcon" :href :html-for="inputId">
+      <slot>{{ label }}</slot>
+    </UiLabel>
+    <div>
+      <VtsIcon :icon accent="current" class="before" />
+      <input
+        :id="inputId"
+        v-model.trim="modelValue"
+        class="typo p1-regular input text-ellipsis"
+        :type
+        :disabled
+        v-bind="attrs"
+      />
+      <VtsIcon
+        v-if="!attrs.disabled && modelValue && clearable"
+        :icon="faXmark"
+        class="after"
+        accent="info"
+        @click="modelValue = ''"
+      />
+    </div>
+    <UiInfo v-if="slots.info || info" :accent>
+      <slot name="info">{{ info }}</slot>
+    </UiInfo>
   </div>
 </template>
 
 <script lang="ts" setup>
 import VtsIcon from '@core/components/icon/VtsIcon.vue'
-import { uniqueId } from '@core/utils/unique-id.util'
+import UiInfo from '@core/components/ui/info/UiInfo.vue'
+import UiLabel from '@core/components/ui/label/UiLabel.vue'
+import { toVariants } from '@core/utils/to-variants.util'
 import type { IconDefinition } from '@fortawesome/fontawesome-common-types'
 import { faXmark } from '@fortawesome/free-solid-svg-icons'
-import { computed, useAttrs } from 'vue'
+import { computed, useAttrs, useId } from 'vue'
+
+type InputAccent = 'info' | 'warning' | 'danger'
+type InputType = 'text' | 'number' | 'password | search'
 
 defineOptions({
   inheritAttrs: false,
 })
 
-defineProps<{
+const { accent, id } = defineProps<{
```

Comment:
You can set a default value directly here:
```suggestion
const { accent, id = useId() } = defineProps<{
```